### PR TITLE
[buildkite] stop using str Enum

### PR DESCRIPTION
**Pull request by alangenfeld**
to enable building the pipeline in newer python versions as well as the current version

https://github.com/python/cpython/issues/100458

## How I Tested These Changes

bk

### CODE REVIEW

File: .buildkite/dagster-buildkite/dagster_buildkite/steps/integration.py [hunk 1]
```diff
@@ -60,12 +64,12 @@ def build_backcompat_suite_steps() -> List[BuildkiteTopLevelStep]:
...
 
-def backcompat_extra_cmds(_, factor: str) -> List[str]:
+def backcompat_extra_cmds(_, factor: Optional[str]) -> List[str]:
     tox_factor_map = {
         "user-code-latest-release": LATEST_DAGSTER_RELEASE,
         "user-code-earliest-release": EARLIEST_TESTED_RELEASE,
     }
-
+    assert factor
     webserver_version = DAGSTER_CURRENT_BRANCH
```
Comment (gibsondan) [inline .buildkite/dagster-buildkite/dagster_buildkite/steps/integration.py:73]: ?

**Review by gibsondan — APPROVED** [review]
see inline

File: .buildkite/dagster-buildkite/dagster_buildkite/steps/integration.py (hunk 1 above)
Comment (alangenfeld) [reply .buildkite/dagster-buildkite/dagster_buildkite/steps/integration.py:73]: aligning the typing [1] makes `factor` `Optional` but we must always be getting it in this context so i just asserted to make the type checker happy

File: .buildkite/dagster-buildkite/dagster_buildkite/steps/integration.py [hunk 2]
```diff
@@ -201,7 +205,7 @@ def build_k8s_suite_steps():
 def build_integration_suite_steps(
     directory: str,
     pytest_tox_factors: Optional[List[str]],
-    pytest_extra_cmds: Optional[Callable] = None,
+    pytest_extra_cmds: Optional[PytestExtraCommandsFunction] = None,
```
Comment (alangenfeld) [inline .buildkite/dagster-buildkite/dagster_buildkite/steps/integration.py:208]: [1]

**Pull request merged**